File: java/src/test/java/io/binac/leetcode/CandyTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CandyTest {
    private static final Candy.Solution1 SOLUTION1 = new Candy.Solution1();

    private final int[] RATINGS1 = {1, 0, 2};
    private final int EXPECTED1 = 5;

    private final int[] RATINGS2 = {1, 2, 2};
    private final int EXPECTED2 = 4;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.candy(RATINGS1));
        assertEquals(EXPECTED2, SOLUTION1.candy(RATINGS2));
    }
}

File: cpp/unittests/CandyTest.cpp
#include "Candy.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(Candy, Solution135_1) {
  std::vector<int> Ratings1{1, 0, 2};
  const int Expected1 = 5;
  EXPECT_EQ(Expected1, Solution135_1().candy(Ratings1));

  std::vector<int> Ratings2{1, 2, 2};
  const int Expected2 = 4;
  EXPECT_EQ(Expected2, Solution135_1().candy(Ratings2));
}